File: library_common_business/src/main/java/cn/jack/library_common_business/service/ApiArticleService.java
package cn.jack.library_common_business.service;

import cn.jack.library_common_business.entiy.ProjectInfoList;
import io.reactivex.Observable;
import jack.retrofit2_rxjava2.model.ApiResponse;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ApiArticleService {

    /**
     * 列表-我的收藏
     */
    @GET("lg/collect/list/{page}/json")
    Observable<ApiResponse<ProjectInfoList>> getArticleCollection(@Path("page") int page);

    /**
     * 收藏文章
     */
    @POST("lg/collect/{id}/json")
    Observable<ApiResponse<String>> collectAtrticle(@Path("id") String id);

    /**
     * 取消收藏
     */
    @POST("lg/uncollect_originId/{id}/json")
    Observable<ApiResponse<String>> unCollectAtrticle(@Path("id") String id);

    /**
     * 知识体系下的文章
     */
    @GET("article/list/{page}/json")
    Observable<ApiResponse<ProjectInfoList>> pageArticleList(@Path("page") int page, @Query("cid") String id);
}

File: library_common_business/src/main/java/cn/jack/library_common_business/service/ApiArticleService.kt
package cn.jack.library_common_business.service

import cn.jack.library_common_business.entiy.ApiResponse
import cn.jack.library_common_business.entiy.ProjectInfoList
import kotlinx.coroutines.flow.Flow
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path
import retrofit2.http.Query

interface ApiArticleService {
    /**
     * 列表-我的收藏
     */
    @GET("lg/collect/list/{page}/json")
    fun getArticleCollection(@Path("page") page: Int): Flow<ApiResponse<ProjectInfoList>>

    /**
     * 收藏文章
     */
    @POST("lg/collect/{id}/json")
    fun collectAtrticle(@Path("id") id: String): Flow<ApiResponse<String>>

    /**
     * 取消收藏
     */
    @POST("lg/uncollect_originId/{id}/json")
    fun unCollectAtrticle(@Path("id") id: String): Flow<ApiResponse<String>>

    /**
     * 知识体系下的文章
     */
    @GET("article/list/{page}/json")
    fun pageArticleList(
        @Path("page") page: Int,
        @Query("cid") id: Int
    ): Flow<ApiResponse<ProjectInfoList>>
}